prod.Dockerfile

```diff
@@ -15,73 +15,122 @@ COPY . .
 RUN pnpm install -f --offline
 
 
-FROM base_builder as backend
-RUN apk add caddy
+FROM base_builder AS backend_builder
 WORKDIR /usr/src/app/packages/hoppscotch-backend
 RUN pnpm exec prisma generate
 RUN pnpm run build
-COPY --from=base_builder /usr/src/app/packages/hoppscotch-backend/backend.Caddyfile /etc/caddy/backend.Caddyfile
+RUN pnpm --filter=hoppscotch-backend deploy /dist/backend --prod
+WORKDIR /dist/backend
+RUN pnpm exec prisma generate
+
+FROM node:20-alpine3.19 AS backend
+RUN apk add caddy
+RUN npm install -g pnpm
+
+RUN addgroup -S hoppgroup && adduser -S hoppuser -G hoppgroup
+
+COPY --from=base_builder  /usr/src/app/packages/hoppscotch-backend/backend.Caddyfile /etc/caddy/backend.Caddyfile
+COPY --from=backend_builder --chown=hoppuser:hoppgroup --chmod=755 /dist/backend /dist/backend
+COPY --from=base_builder --chown=hoppuser:hoppgroup --chmod=755 /usr/src/app/packages/hoppscotch-backend/prod_run.mjs /dist/backend
+
 # Remove the env file to avoid backend copying it in and using it
-RUN rm "../../.env"
 ENV PRODUCTION="true"
 ENV PORT=8080
 ENV APP_PORT=${PORT}
 ENV DB_URL=${DATABASE_URL}
-CMD ["node", "/usr/src/app/packages/hoppscotch-backend/prod_run.mjs"]
+
+USER hoppuser
+
+WORKDIR /dist/backend
+
+CMD ["node", "prod_run.mjs"]
 EXPOSE 80
 EXPOSE 3170
 
-FROM base_builder as fe_builder
+FROM base_builder AS fe_builder
 WORKDIR /usr/src/app/packages/hoppscotch-selfhost-web
 RUN pnpm run generate
 
-FROM caddy:2-alpine as app
+FROM caddy:2-alpine AS app
 WORKDIR /site
-COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/prod_run.mjs /usr
-COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/selfhost-web.Caddyfile /etc/caddy/selfhost-web.Caddyfile
-COPY --from=fe_builder /usr/src/app/packages/hoppscotch-selfhost-web/dist/ .
+RUN addgroup -S hoppgroup && adduser -S hoppuser -G hoppgroup
+
+COPY --from=fe_builder --chown=hoppuser:hoppgroup --chmod=755 /usr/src/app/packages/hoppscotch-selfhost-web/prod_run.mjs /usr
+COPY --from=fe_builder --chown=hoppuser:hoppgroup --chmod=755 /usr/src/app/packages/hoppscotch-selfhost-web/selfhost-web.Caddyfile /etc/caddy/selfhost-web.Caddyfile
+COPY --from=fe_builder --chown=hoppuser:hoppgroup --chmod=755 /usr/src/app/packages/hoppscotch-selfhost-web/dist/ .
+
 RUN apk add nodejs npm
+
 RUN npm install -g @import-meta-env/cli
+
+USER hoppuser
+
 EXPOSE 80
 EXPOSE 3000
 CMD ["/bin/sh", "-c", "node /usr/prod_run.mjs && caddy run --config /etc/caddy/selfhost-web.Caddyfile --adapter caddyfile"]
 
-FROM base_builder as sh_admin_builder
+FROM base_builder AS sh_admin_builder
 WORKDIR /usr/src/app/packages/hoppscotch-sh-admin
 # Generate two builds for `sh-admin`, one based on subpath-access and the regular build
 RUN pnpm run build --outDir dist-multiport-setup
 RUN pnpm run build --outDir dist-subpath-access --base /admin/
 
-FROM caddy:2-alpine as sh_admin
+FROM caddy:2-alpine AS sh_admin
```

Comment:
![image](https://github.com/user-attachments/assets/af783242-f367-40fc-bc1e-d79425bde654)

Same error here too.

Reply:
should be fixed now. ptal.